packages/components/src/test/properties/inputs/UniquePropertyValuesSelector.test.tsx

```diff
@@ -772,6 +773,93 @@ describe("UniquePropertyValuesSelector", () => {
 
       // expect "clear" button to not be shown
       await waitFor(() => expect(container.querySelector(".clear-indicator")).to.be.null);
+
+      const closeFilteringDialogButton = container.querySelector(".presentation-instance-filter-dialog-close-button");
+      await user.click(closeFilteringDialogButton!);
+    });
+
+    it("loads second page when first page is already loaded and empty", async () => {
+      const pageItems = [];
+      for (let i = 0; i < UNIQUE_PROPERTY_VALUES_BATCH_SIZE; i++) {
+        const name = `SkippedValue${i}`;
+        pageItems.push({ displayValue: name, groupedRawValues: [name] });
+      }
+
+      getDistinctValuesIteratorStub.withArgs(matchPageStart(0)).resolves({
+        total: UNIQUE_PROPERTY_VALUES_BATCH_SIZE,
+        items: createAsyncIterator(pageItems),
+      });
+      getDistinctValuesIteratorStub.withArgs(matchPageStart(UNIQUE_PROPERTY_VALUES_BATCH_SIZE)).resolves({
+        total: 2,
+        items: createAsyncIterator([
+          { displayValue: "SkippedValue", groupedRawValues: ["SkippedValue"] },
+          { displayValue: "SearchedValue", groupedRawValues: ["SearchedValue"] },
+        ]),
+      });
+
+      const { getByText, queryByText, queryAllByText, user, container } = render(
+        <TestComponentWithPortalTarget>
+          <UniquePropertyValuesSelector property={propertyDescription} onChange={() => {}} imodel={testImodel} descriptor={descriptor} />,
+        </TestComponentWithPortalTarget>,
+      );
+
+      // open menu
+      const menuSelector = await waitFor(() => getByText("unique-values-property-editor.select-values"));
+      await user.click(menuSelector);
+
+      // type in search
+      const searchSelector = await waitFor(() => container.querySelector(".presentation-async-select-values-container"));
+      await user.type(searchSelector!, "Searched");
+
+      // ensure all searched for values are shown
+      await waitFor(() => {
+        expect(queryByText("SearchedValue")).to.not.be.null;
+        expect(queryAllByText(/SkippedValue/)).to.be.empty;
+      });
+      expect(getDistinctValuesIteratorStub).to.be.calledTwice;
+    });
+
+    it("correctly determines `hasMore` value when fetched display value is undefined", async () => {
+      const pageItems = [
+        { displayValue: "SkippedValue", groupedRawValues: ["SkippedValue"] },
+        { displayValue: undefined, groupedRawValues: [] },
+      ];
+      for (let i = 2; i < UNIQUE_PROPERTY_VALUES_BATCH_SIZE; i++) {
+        const name = `SkippedValue${i}`;
+        pageItems.push({ displayValue: name, groupedRawValues: [name] });
+      }
+      getDistinctValuesIteratorStub.onCall(0).resolves({
+        total: 0,
+        items: createAsyncIterator([]),
+      });
```

Comment:
Are the real cases where first request returns 0 items, and second returns something?

Reply:
No, the first call was only used to not cause more pages to be loaded. I changed it to return a single value instead. It is still not accurate to real cases as an entire page would be loaded, however that causes flakiness in tests because for some reason `AsyncPaginate` sometimes believes the menu is being continuously scrolled and starts loading all the pages prematurely.